src/PixelFlags.cc

```diff
@@ -57,87 +59,153 @@ class FootprintBits : public afw::detection::FootprintFunctor<MaskedImageT> {
+        center = measRecord.getCentroid(); 
+        //  Catch NAN in centroid estimate
+        if (lsst::utils::isnan(center.getX()) || lsst::utils::isnan(center.getY())) {
+            throw LSST_EXCEPT(pex::exceptions::RuntimeError,"Center point passed to PixelFlagsAlgorithm is Nan");
+        }
+    }
+    else {
+        // Attempting to set pixel flags with no centroider, the center will
+        // be determined though the peak pixel
+        PTR(afw::detection::Footprint) footprint = measRecord.getFootprint();
+        // If there is no footprint or the footprint contains no peaks, throw
+        // a runtime error.
+        if (!footprint || footprint->getPeaks().empty()){
+            throw LSST_EXCEPT(pex::exceptions::RuntimeError,"No footprint, or no footprint peaks detected");
+        }
+        else {
+            center.setX(footprint->getPeaks().front().getFx());
+            center.setY(footprint->getPeaks().front().getFy());
+        }
     }
-//  Catch centroids off the image
+
+    //  Initialize bitmask to use in try statments
+    lsst::afw::image::MaskPixel bitmask = 0x0;
+
+    //  Catch centroids off the image
     if (!mimage.getBBox().contains(afw::geom::Point2I(center))) {
-       _flagHandler.setValue(measRecord, EDGE, true);
+        measRecord.set(_offImageKey, true);
+        auto iter = _predefinedAnyKeys.find("EDGE");
+        assert(iter != _predefinedAnyKeys.end());
+        measRecord.set(iter->second, true);
     }
+
     // Check for bits set in the source's Footprint
     afw::detection::Footprint const & footprint(*measRecord.getFootprint());
     func.apply(footprint);
-    if (func.getBits() & (MaskedImageT::Mask::getPlaneBitMask("EDGE") |
-                          MaskedImageT::Mask::getPlaneBitMask("NO_DATA"))) {
-        _flagHandler.setValue(measRecord, EDGE, true);
-    }
-    if (func.getBits() & MaskedImageT::Mask::getPlaneBitMask("BAD")) {
-        _flagHandler.setValue(measRecord, BAD, true);
-    }
-    if (func.getBits() & MaskedImageT::Mask::getPlaneBitMask("INTRP")) {
-        _flagHandler.setValue(measRecord, INTERPOLATED, true);
-    }
-    if (func.getBits() & MaskedImageT::Mask::getPlaneBitMask("SAT")) {
-        _flagHandler.setValue(measRecord, SATURATED, true);
+
+    // Check to make sure the NO_DATA mask exists
+    try {
+        bitmask = MaskedImageT::Mask::getPlaneBitMask("NO_DATA");
+    } catch (pex::exceptions::InvalidParameterError & err){
+        throw LSST_EXCEPT(FatalAlgorithmError, err.what());
     }
-    if (func.getBits() & MaskedImageT::Mask::getPlaneBitMask("CR")) {
-        _flagHandler.setValue(measRecord, CR, true);
+
+    // Set the EDGE flag if the bitmask has NO_DATA set
+    if (func.getBits() & MaskedImageT::Mask::getPlaneBitMask("NO_DATA")){
+        auto iter = _predefinedAnyKeys.find("EDGE");
+        assert(iter != _predefinedAnyKeys.end());
+        measRecord.set(iter->second, true);
     }
```

Comment:
As earlier, do you need the `try... catch` followed by the `if`? It seems like you could just surround the latter in the exception handler, and -- in the process -- eliminate `bitmask` entirely.

Also, again, these three lines could become one.